### PR TITLE
misc: lint irdl.py

### PR DESCRIPTION
**Pull request by superlopuh**
In celebration of the pdl working, I decided to lint irdl.py.

The changes are mostly:

changing lambdas to functions
adding `# type: ignore` when there is no hope of actually typing things properly

2 files, 30 errors to go!

### CODE REVIEW

**Review by georgebisbas — APPROVED** [review]

File: xdsl/irdl.py [hunk 1]
```diff
@@ -257,7 +257,8 @@ def irdl_to_attr_constraint(
         generic_args = ()
 
         # Get the Generic parent class to get the TypeVar parameters
-        for parent in origin.__orig_bases__:  # type: ignore
+        bases = cast(Sequence[Any], origin.__orig_bases__)  # type: ignore
```
Comment (math-fehr) [inline xdsl/irdl.py:260]: Just a sanity check, do we still need the type: ignore here?
Or is it because it doesn't know about `__orig_bases__`?

**Review by math-fehr — APPROVED** [review]
Nice!

**Review by webmiche — APPROVED** [review]
LGTM!

File: xdsl/irdl.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/irdl.py:260]: Unfortunately yes, although weirdly only for `pyright`, not Pylance in vscode. There are two errors on this line, the fact that `origin` is unknown, and the fact that `origin.__orig_bases__` is unknown. The `type: ignore` here is for the first one, and the cast is for the second

**Pull request merged**